### DISCUSSION TITLE
[Feature] Mobile Multi Server Switching

### DISCUSSION THREAD

**Discussion by cwerkey** [immich-app/immich#7700] [Feature Request]
### The feature

It would be nice to be able to have the option to swap between multiple server instances without having to remember the url of each one and have to enter it every time. A list of saved/know URL’s that would have an option to store/recall the username and/or the password for simpler swapping. Even if it was just a list of stored/previous connections you could select. 

Use Case: I currently have a test immich instance for playing with new features and testing how the mechanics work with stock photos, and a production immich instance running with real users and photos shared to family
I could also see a use case if you host a personal server and also share a friends. 

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
Related: #2335

**Reply by SuBra98** [reply]
Hello, I feel #2335 addresses a different need. Like the next replyer responds, the request is a feature just like Nextcloud offers, to add multiple accounts and be able to switch between those accounts. Starting point is tht there are multiple different servers.

**Comment by NullEnt1ty** [comment]
Another use case for me is that me and a friend are both using separate Immich instances. I'd like to use the app with his and mine instance.

You could take a look at Nextcloud's app for example, which allows the user to add multiple accounts and switch between them.

**Comment by joeskeen** [comment]
Why is this closed?

Ideally for me this would work more like Plex, where you can be logged into multiple servers simultaneously (search retrieves results from all servers). I know that the main difference here is that Immich accounts are per server not global, but I still think this should be possible.

**Comment by Hritik14** [comment]
Can we reopen this ? Multiple server support sounds really good to have.

**Comment by lkmill** [comment]
+1 for opening this again. I want to easily be able to switch between my own, my families and my friends immich server instances.

**Comment by KBrygger** [comment]
Joining this request. I also see a great need for this feature.

**Comment by matteohoeren** [comment]
Yes, please do re-open this request! I have an instance for myself, my family and a couple of friends invited me to join their instances. 
Maybe Immich could also think about a federation feature to share albums/folders with other instances. That would be amazing!
I know that this is not trivial to pull off :)

**Comment by Zilusion** [comment]
It would be very useful to support multiple server accounts in the mobile app, with saved connections and quick switching between them. A common use case is having one personal/family Immich server and another separate instance for a different library, testing. Re-entering the server URL and credentials every time is unnecessary friction. This should work more like Nextcloud, where multiple accounts can be added and switched between easily.

**Comment by createchange** [comment]
Adding my vote for this.

I recently stood up a completely private instance which contains all my photos. I stood up a second server which I selectively share photos/video with friends/family.

It would be nice to be able to bounce between the servers seamlessly. I do recognize that of all the people I just named that I will be sharing with, I am the only one that will be impacted by this. So I do understand it's kind of a niche issue, but evidently (as per the above, including votes) it's not _that_ niche.

Thanks for the great product! Honestly, fantastic.

**Comment by qguv** [comment]
Please re-open this! I have folders on my phone which should auto-upload to different immich accounts, one for work and one personal one. The nextcloud app supports auto-upload to two different servers, but immich doesn't. As I'm using immich to replace nextcloud photos, I'd really appreciate this feature as it would mean that the switch wouldn't cause this regression in functionality, especially since immich is such an improvement over nextcloud photos in the first place.

Thanks!!

**Comment by ypnos** [comment]
It's incredible how popular Immich has become. I now have three servers that I frequently upload pictures to, one of them being my own. 

I would greatly appreciate being able to use the mobile app for all of them.
